proposals/0031-hlsl-vector-matrix-operations.md

```diff
@@ -160,7 +160,58 @@ TBD
 
 ### Builtins
 
-TBD
+Although these "builtins" are not intended to be part of the HLSL language, and
+no promises are made that these will continue to be available over time, this
+proposal describes an implementation in terms of builtins such as this. For
+this reason it is useful to have them described here as a reference point.
+
+Each builtin corresponds to one of the operations described in [0029].
+
+```c++
+namespace dx {
+namespace linalg {
+namespace details {
+
+// dx.op.matvecmul
+template <typename TYo, int NUMo, typename TYi, int NUMi, typename RES>
+void __builtin_MatVecMul(vector<TYi, NUMi> InputVector,
```

Comment:
Make the out vector arg in Mul and MulAdd as the first in sync with the implementation